### ISSUE TITLE
Job feasibility checks

### ISSUE THREAD

**Issue by trws** [flux-framework/flux-core#269]
This may need to be cross-listed to flux-sched.  Pending alternate access to the RDL, the scheduler needs to provide feasibility feedback on job submission.

**Comment by dongahn** [comment]
Perhaps the scheduler framework service can query the max resource shape information from RDL right after initializing the RDL. Then, on the state transition, RESERVED -> SUBMITTED, it can call a feasibility check action to do this check? Initially, this action will only check simple max resource  satisfiability, but later this can be extended to cover site constraints etc as well. 

Another option would be to have "satisfiability check" module and have our submit API gets this check before bothering waking up the scheduler modeul. 

BTW, the current scheduler state transition is PENDING -> SCHEDREQ -> SELECTED on successful job allocation; PENDING->SCHEREQ->PENDING on unsuccessful allocation. You don't need to be notified of the latter failed scheduling attempt, do you?

**Comment by trws** [comment]
I don't think I'll need unsuccessful allocation.  At this stage, all I really need is to be able to give reasonable feedback when the user requests something that is impossible.   Come to think of it, the session should also populate a default RDL with node/core information from its bootstrap if no RDL is specified... I'll put in a new ticket for that, since it's something we can do relatively easily that will make this all rather easier to use and test.

**Comment by dongahn** [comment]
I like it! If the well-formed RDL gets populated in the bootstrapped environment, the scheduler service also can load that in and use it for scheduling (and everything becomes consistent). This will make testing whole lot easier!
